test/http_middleware_test.lua

```diff
@@ -162,6 +163,24 @@ g.test_v1_wrong_handler = function()
     )
 end
 
+g.test_v1_handler_raise_an_error = function()
+    local request = {endpoint = table.copy(route)}
+    local observer = stub_observer(function() end)
+    local handler = function()
+        error('Handler is broken')
```

Comment:
The original issue is about error thrown with callback https://github.com/tarantool/metrics/issues/328. Does PR cover it? Or the case mentioned in the issue is "/metrics handler wrapped in HTTP middleware"?

Reply:
Well, based on repo search, the only place where you can get "incorrect http handler" error is here, so it should fix the behavior. Error thrown on invoke callback is not pcalled anywhere but here.